### PR TITLE
CC-4419 Skipping Elasticsearch export on delegate hosts

### PR DESCRIPTION
**Pull request by momelchuk**
Fixes CC-4419

Added condition for skipping elasticsearch migrations if
the delegate host doesn't have Elasticsearch cluster
configuration for CC

### CODE REVIEW

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by webhookbotuser** [comment]
Can one of the admins verify this patch?

**Comment by DeKapito** [comment]
ok to test

File: pkg/rpm/postupgrade [hunk 1]
```diff
@@ -139,16 +139,25 @@ docker run --rm -d --name $SVC_NAME -p 9200:9200 \
   sh -c "$ELASTIC_BIN -f -Des.node.name=$NODE_NAME \
   -Des.cluster.name=$CLUSTER_NAME"
 
-if check_elasticsearch; then
-    report "SUCCESS" "Container started within timeout"
+#
+# CC-4419: If Elasticsearch configuration file and data folder does not exist
+# on the delegate that migration shouldn't start
+#
+if [[ ! -d $HOST_ES_DATA_DIR ] || [ ! $CLUSTER_NAME ]]; then
```
Comment (jpeacock-zenoss) [inline pkg/rpm/postupgrade:146]: If this is supposed to run in a bash shell, this syntax won't work.  It should be
```
if [[ ! -d $HOST_ES_DATA_DIR || ! $CLUSTER_NAME ]]; then
```

Comment (jpeacock-zenoss) [reply pkg/rpm/postupgrade:146]: Example:
```
$ if [[ ! -d /blah ] || [ ! "" ]]; then echo "hi"; fi
-bash: syntax error in conditional expression
-bash: syntax error near `]'
$ if [[ ! -d /blah || ! "" ]]; then echo "hi"; fi
hi
```

**Review by jpeacock-zenoss — CHANGES_REQUESTED** [review]

File: pkg/rpm/postupgrade [hunk 2]
```diff
@@ -139,16 +139,25 @@ docker run --rm -d --name $SVC_NAME -p 9200:9200 \
   sh -c "$ELASTIC_BIN -f -Des.node.name=$NODE_NAME \
   -Des.cluster.name=$CLUSTER_NAME"
 
-if check_elasticsearch; then
-    report "SUCCESS" "Container started within timeout"
+#
```
Comment (Deer-WarLord) [inline pkg/rpm/postupgrade:142]: ```cp -ar $HOST_ISVCS_DIR/elasticsearch-serviced $HOST_ISVCS_DIR/elasticsearch-serviced.backup```
ES backup shouldn't run on the delegate host

```
Starting docker container elasticsearch-serviced ...
```
shouldn't run on the delegate host

File: pkg/rpm/postupgrade [hunk 3]
```diff
@@ -139,16 +139,25 @@ docker run --rm -d --name $SVC_NAME -p 9200:9200 \
...
+    fi
+
+    if migrate_elasticsearch; then
+        report "SUCCESS" "Export completed"
+    else
+        report "FAILURE" "Export failed try make the export manual"
+    fi
 fi
 
 echo "Stopping the container with old elasticsearch"
```
Comment (Deer-WarLord) [inline pkg/rpm/postupgrade:163]: Stop the ES container, shouldn't run on the delegate host

File: pkg/rpm/postupgrade [hunk 4]
```diff
@@ -175,15 +184,24 @@ docker run --rm -d --name $SVC_NAME -p 9200:9200 \
   su elastic -c "$ELASTIC_BIN -Ecluster.initial_master_nodes=$NODE_NAME \
   -Enode.name=$NODE_NAME -Ecluster.name=$CLUSTER_NAME"
 
-if check_elasticsearch; then
-    report "SUCCESS" "Container started within timeout"
+#
```
Comment (Deer-WarLord) [inline pkg/rpm/postupgrade:187]: 1) Preparing the host for data import to new elasticsearch storage
2) Starting container with new elasticsearch
Shouldn't run on the delegate host

File: pkg/rpm/postupgrade [hunk 5]
```diff
@@ -175,15 +184,24 @@ docker run --rm -d --name $SVC_NAME -p 9200:9200 \
...
+        report "FAILURE" "Container failed to start within 120 seconds"
+    fi
+
+    echo "Importing data to new elasticsearch"
+    $HOME_SERVICED/bin/elastic -i -c $HOME_SERVICED/etc/es_cluster.ini \
+      -f $HOST_ISVCS_DIR/elasticsearch_data.json
+fi
 
 echo "Stopping the container with new elasticsearch"
 docker stop $SVC_NAME
```
Comment (Deer-WarLord) [inline pkg/rpm/postupgrade:207]: Shouldn't run on the delegate host

File: pkg/rpm/postupgrade (hunk 1 above)
Comment (momelchuk) [reply pkg/rpm/postupgrade:146]: yeah. my fail

File: pkg/rpm/postupgrade [hunk 6]
```diff
@@ -127,66 +127,84 @@ fi
...
-chown 1001:1001 -R $HOST_ES_DATA_DIR
-
 echo "Setting persistent vm.max_map_count in the sysctl.conf"
 echo vm.max_map_count=262144 >> /etc/sysctl.conf
 sysctl --system
+#
+# CC-4419: If Elasticsearch configuration file and data folder does not exist
+# on the delegate that migration shouldn't start
+#
+if [[ ! -d $HOST_ES_DATA_DIR ] || [ -z $CLUSTER_NAME ]]; then
```
Comment (jpeacock-zenoss) [inline pkg/rpm/postupgrade:177]: This has extra square braces.  The form should be `[[ ... || ... ]]` not `[[ ... ] || [ ... ]]`.

File: pkg/rpm/postupgrade [hunk 7]
```diff
@@ -127,66 +127,84 @@ fi
...
-    report "FAILURE" "Container failed to start within 120 seconds"
-fi
-
-if migrate_elasticsearch; then
-    report "SUCCESS" "Export completed"
+#
+# CC-4419: If Elasticsearch configuration file and data folder does not exist
+# on the delegate that migration shouldn't start
+#
+if [[ ! -d $HOST_ES_DATA_DIR ] || [ -z $CLUSTER_NAME ]]; then
```
Comment (jpeacock-zenoss) [inline pkg/rpm/postupgrade:137]: This has extra square braces. The form should be [[ ... || ... ]] not [[ ... ] || [ ... ]].

**Review by jpeacock-zenoss — CHANGES_REQUESTED** [review]

**Review by Deer-WarLord — APPROVED** [review]

**Review by jpeacock-zenoss — APPROVED** [review]

**Pull request closed without merging**